cmd/kubeadm/app/util/apiclient/wait.go

```diff
@@ -18,20 +18,21 @@ package apiclient
 
 import (
 	"fmt"
+	"io"
 	"net/http"
 	"time"
 
 	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 	clientset "k8s.io/client-go/kubernetes"
-	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
 )
 
 // WaitForAPI waits for the API Server's /healthz endpoint to report "ok"
-func WaitForAPI(client clientset.Interface) {
+func WaitForAPI(client clientset.Interface, timeout time.Duration) error {
 	start := time.Now()
-	wait.PollInfinite(kubeadmconstants.APICallRetryInterval, func() (bool, error) {
```

Comment:
I don't think so; as it often takes  ~15 secs for control plane to come up
Whether we start polling immediately or after 0.5secs doesn't seem to matter.